core/src/test/java/apoc/index/SchemaIndexTest.java

```diff
@@ -86,76 +102,191 @@ public void testDistinctPropertiesOnFirstIndex() {
+        
+        db.executeTransactionally("drop index VoidIndex");
     }
 
     @Test(timeout = 5000L)
-    public void testDistinctWithFullTextIndexShouldNotHangs() throws Exception {
-        db.executeTransactionally("CREATE FULLTEXT INDEX fulltextLabel1 FOR (n:Label1) ON EACH [n.prop1]");
+    public void testDistinctWithCompositeIndexShouldNotHangs() throws Exception {
+        db.executeTransactionally("create index EmptyLabel for (n:EmptyLabel) on (n.one)");
+        db.executeTransactionally("create index EmptyCompositeLabel for (n:EmptyCompositeLabel) on (n.two, n.three)");
 
-        db.executeTransactionally("CALL db.awaitIndexes()");
+        testCall(db, "CALL apoc.schema.properties.distinct($label, $key)",
+                map("label", "EmptyLabel", "key", "one"),
+                row -> assertEquals(emptyList(), row.get("value"))
+        );
 
         testCall(db, "CALL apoc.schema.properties.distinct($label, $key)",
-                map("label", "Label1", "key", "prop1"), 
-                row -> assertEquals(Set.of("AA", "EE", "UU", "Ryan", "Michael"), Set.copyOf((List)row.get("value"))) 
+                map("label", "EmptyCompositeLabel", "key", "two"),
+                row -> assertEquals(emptyList(), row.get("value"))
         );
 
-        testResult(db, "CALL apoc.schema.properties.distinctCount($label, $key)",
-                map("label", "Label1", "key", "prop1"),
+        db.executeTransactionally("drop index EmptyLabel");
+        db.executeTransactionally("drop index EmptyCompositeLabel");
+    }
+
+    @Test(timeout = 5000L)
+    public void testDistinctWithCompositeIndexWithMixedRepeatedProps() throws Exception {
+        db.executeTransactionally("create index SchemaTest for (n:SchemaTest) on (n.prop1, n.prop2)");
+        db.executeTransactionally("CALL db.awaitIndexes()");
+
+        testResult(db, SCHEMA_DISTINCT_COUNT_ORDERED,
+                map("label", SCHEMA_LABEL, "key", "prop2"),
                 res -> {
-                    assertDistinctCountProperties("Label1", "prop1", List.of("AA", "EE", "UU"), () -> 1L, res);
-                    assertDistinctCountProperties("Label1", "prop1", List.of("Ryan"), () -> 3L, res);
-                    assertDistinctCountProperties("Label1", "prop1", List.of("Michael"), () -> 1L, res);
+                    assertDistinctCountProperties(SCHEMA_LABEL, "prop2", List.of("bar"), 2L, res);
+                    assertDistinctCountProperties(SCHEMA_LABEL, "prop2", List.of("foo"), 1L, res);
                     assertFalse(res.hasNext());
                 });
 
-        db.executeTransactionally("DROP INDEX fulltextLabel1");
+        testCall(db, "CALL apoc.schema.properties.distinct($label, $key)",
+                map("label", SCHEMA_LABEL, "key", "prop2"),
+                row -> assertEquals(Set.of("bar", "foo"), Set.copyOf((List)row.get("value")))
+        );
+
+        testResult(db, SCHEMA_DISTINCT_COUNT_ORDERED,
+                map("label","","key",""),
```

Comment:
So if the label or property is an empty string, that means all labels/properties? I think it would make sense to add that to the documentation as well, can add that to the existing documentation card

Reply:
Sounds good, I added it [here](https://trello.com/c/MBiu8YIz/1827-improve-documentation-about-apocschemapropertiesdistinctcount) as a checklist.